.github/workflows/nix.yaml

```
@@ -114,25 +111,22 @@ jobs:
 
       - name: Build Docker image
         id: docker
-        if: steps.check-conditions.outputs.should_build == 'true' && matrix.system == 'x86_64-linux' && needs.check-conditions.outputs.should-push-docker == 'true'
+        if: steps.check-conditions.outputs.should_build == 'true' && contains(matrix.system, 'linux') && needs.check-conditions.outputs.should-push-docker == 'true'
         run: |
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC2086:info:1:22: Double quote to prevent globbing and word splitting

(shellcheck)